share/spack/docker/ci/Dockerfile

```diff
@@ -0,0 +1,74 @@
+ARG BASE
+
+FROM $BASE
+MAINTAINER Spack Maintainers <maintainers@spack.io>
+
+ENV CURRENTLY_BUILDING_DOCKER_IMAGE=1
+
+MASK PUSH
+MASK [[ $DISTRO == centos ]]
+RUN yum update -y                                          \
+ && yum install -y gmp-devel libmpc-devel mpfr-devel which \
+ && rm -rf /var/cache/yum                                  \
+ && yum clean all
+
+MASK [[ $DISTRO == ubuntu ]]
+RUN apt-get -yqq update                                     \
+ && apt-get -yqq install clang g++-5 gcc-5 gfortran-5 unzip \
+ && rm -rf /var/lib/apt/lists/*
+
+
+MASK [[ $DISTRO == centos ]]
+# Download, build and install gcc 5.5.0
+RUN mkdir -p /gcc-bld /opt/gcc/5.5.0                                    \
+ && ( cd /                                                              \
+ &&   curl -OL https://ftp.gnu.org/gnu/gcc/gcc-5.5.0/gcc-5.5.0.tar.xz   \
+ &&   tar -xvf gcc-5.5.0.tar.xz                                         \
+ &&   cd gcc-bld                                                        \
+ &&   ../gcc-5.5.0/configure                                            \
+          --enable-languages=c,c++,fortran                              \
+          --disable-multilib                                            \
+          --prefix=/opt/gcc/5.5.0                                       \
+ &&   make -j\$(nproc)                                                  \
```

Comment:
Just out of curiosity, is that an escape you put in there before the `$`?  What's that doing?

Reply:
The Dockerfiles are preprocessed with the `dpp.bash` shell script.  This [preprocessing](https://github.com/spack/spack/blob/444c286e6225a30594dc4e3e1da10baf2fbc51b6/share/spack/docker/dpp.bash#L40) includes (among other things) expansion of any shell variables and inline commands.

So in cases like above, where I want docker to actually run `make -j$(nproc)`, I would need to escape the `$`.  Otherwise, `nproc` would run on the host side and the result would be substituted verbatim into the resulting Dockerfile.